components/feral-watchdog/gpu.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+const (
+	REBOOT_DELAY = 15 * time.Second
+)
+
+type GPUHandler struct {
+	mu              sync.Mutex
+	logger          *zap.Logger
+	commandHandler  *CommandHandler
+	rebootTimer     *time.Timer
+	rebootScheduled bool
+}
+
+func NewGPUHandler(logger *zap.Logger, commandHandler *CommandHandler) *GPUHandler {
+	return &GPUHandler{
+		logger:          logger,
+		commandHandler:  commandHandler,
+		rebootScheduled: false,
+	}
+}
+
+func (g *GPUHandler) scheduleGPUReboot(ctx context.Context) {
+	g.mu.Lock()
+	defer g.mu.Unlock()
+
+	// If a reboot is already scheduled, ignore this request
+	if g.rebootScheduled {
+		g.logger.Info("GPU: reboot already scheduled, ignoring request")
+		return
+	}
+
+	g.logger.Info("GPU: scheduling reboot")
+	g.rebootScheduled = true
+
+	// Create a timer to reboot after 15 seconds
+	g.rebootTimer = time.AfterFunc(REBOOT_DELAY, func() {
```

Comment:
This should be stopped whenever the context is canceled. So my suggestion is to add a function to release all pending resources.